web/src/main/java/no/nav/foreldrepenger/web/app/tjenester/fpoversikt/FpOversiktDtoTjeneste.java

```diff
@@ -69,15 +80,126 @@ public Sak hentSak(String saksnummer) {
+        return mottatteSøknader.stream()
+            .map(md -> {
+                var status = statusForSøknad(åpenYtelseBehandling, md);
+                var perioder = ytelseFordelingTjeneste.hentAggregatHvisEksisterer(md.getBehandlingId()).map(ytelseFordelingAggregat -> {
+                    var oppgittFordeling = ytelseFordelingAggregat.getOppgittFordeling();
+                    return oppgittFordeling.getPerioder().stream().map(p -> tilDto(p)).collect(Collectors.toSet());
+                }).orElse(Set.of());
+                return new FpSak.Søknad(status, md.getMottattTidspunkt(), perioder);
+            })
+            .filter(s -> !s.perioder().isEmpty()) //Filtrerer ut søknaden som ikke er registert i YF. Feks behandling står i papir punching
+            .collect(Collectors.toSet());
+    }
+
+    private static SøknadStatus statusForSøknad(Optional<Behandling> åpenYtelseBehandling, MottattDokument md) {
+        return åpenYtelseBehandling.filter(b -> b.getId().equals(md.getBehandlingId())).map(b -> SøknadStatus.MOTTATT).orElse(SøknadStatus.BEHANDLET);
+    }
+
+    private static FpSak.Søknad.Periode tilDto(OppgittPeriodeEntitet periode) {
+        return new FpSak.Søknad.Periode(periode.getFom(), periode.getTom());
+    }
+
+    private static Set<SvpSak.Søknad> finnSvpSøknader(Optional<Behandling> åpenYtelseBehandling, List<MottattDokument> mottatteSøknader) {
+        return mottatteSøknader.stream()
+            .map(md -> {
+                var status = statusForSøknad(åpenYtelseBehandling, md);
+                return new SvpSak.Søknad(status, md.getMottattTidspunkt());
+            })
+            .collect(Collectors.toSet());
+    }
+
+    private static Set<EsSak.Søknad> finnEsSøknader(Optional<Behandling> åpenYtelseBehandling, List<MottattDokument> mottatteSøknader) {
+        return mottatteSøknader.stream()
+            .map(md -> {
+                var status = statusForSøknad(åpenYtelseBehandling, md);
+                return new EsSak.Søknad(status, md.getMottattTidspunkt());
+            })
+            .collect(Collectors.toSet());
+    }
+
+    private List<Behandling> finnIkkeHenlagteBehandlinger(Fagsak fagsak) {
+        return behandlingRepository.hentAbsoluttAlleBehandlingerForFagsak(fagsak.getId());
+    }
+
+    private Set<Sak.Aksjonspunkt> finnAksjonspunkt(List<Behandling> ikkeHenlagteBehandlinger) {
+        return ikkeHenlagteBehandlinger.stream().flatMap(b -> b.getAksjonspunkter().stream())
+            .filter(a -> a.erOpprettet() || a.erUtført())
+            .map(a -> {
+                var status = a.erOpprettet() ? Sak.Aksjonspunkt.Status.OPPRETTET : Sak.Aksjonspunkt.Status.UTFØRT;
+                var venteÅrsak = Venteårsak.UDEFINERT.equals(a.getVenteårsak()) ? null : a.getVenteårsak().getKode();
+                return new Sak.Aksjonspunkt(a.getAksjonspunktDefinisjon().getKode(), status, venteÅrsak, a.getOpprettetTidspunkt());
```

Comment:
Hm, hvilke AP trenger du vite om? Blir litt fuzzy å vedlikeholde streng-koder på lang sikt. Ser det i LOS.

Reply:
Ditto for venteårsak. Synes vi skal være forsiktig med getKode i dette landskapet :-)

Reply:
Ja tenkte på det, men nesten like vanskelig å vedlikeholde enum for alle sammen. Akkurat nå trenger vi bare noen av AP med venteårsaker, men litt usikker på hva vi trenger senere (var snakk om noe avvik i beregning). Så tenker det er greit å lagre ned alle, både opprettet og utført

Reply:
Hm, går det ikke an å lese ut av BG om det har vært skjønnsfastsettelse?

Reply:
Cecilie mente ikke det, men skal dobbeltsjekke med petter

Reply:
Det tryggeste er nok å se på aksjonspunkt hvis du vil se saker der de faktisk har skjønnsfastsatt. Hvis du bare ser på grunnlaget vil senere behandlinger ha data som tilsier at de er skjønnsfastsatt, mens de i realiteten bare har fått kopiert grunnlaget fra forrige behandling (ved start i uttak f.eks)

Reply:
fikset nå. Sender bare over opprettet ventepunkter med venteårsak + frist